python/hail/dataset.py

```diff
@@ -3068,13 +3069,15 @@ def lmmreg(self, kinshipMatrix, y, covariates=[], global_root="global.lmmreg", v
 
         This plan can be modified as follows:
 
-        - Set ``run_assoc=False`` to not test any variants for association, i.e. skip Step 5.
-        - Set ``use_ml=True`` to use maximum likelihood instead of REML in Steps 4 and 5.
-        - Set the ``delta`` argument to manually set the value of :math:`\delta` rather that fitting :math:`\delta` in Step 4.
-        - Set the ``global_root`` argument to change the global annotation root in Step 4.
-        - Set the ``va_root`` argument to change the variant annotation root in Step 5.
+        - Set ``run_assoc=False`` to not test any variants for association, i.e. skip Step 4.
+        - Set ``use_ml=True`` to use maximum likelihood instead of REML in Steps 3 and 4.
+        - Set the ``delta`` argument to manually set the value of :math:`\delta` rather that fitting :math:`\delta` in Step 3.
+        - Set the ``global_root`` argument to change the global annotation root in Step 3.
+        - Set the ``va_root`` argument to change the variant annotation root in Step 4.
+        - Set the ``filter_variants_expr`` argument to filter the dataset to only those variants for which variant association should be done.
```

Comment:
It seems like fitting the global parameters should be a separate method because it produces a more flexible language for the users. What would need to be done to achieve this?